components/core/src/clp_s/timestamp_parser/TimestampParser.cpp

```diff
@@ -0,0 +1,382 @@
+                    return ErrorCodeEnum::IncompatibleTimestampPattern;
+                }
+
+                parsed_day = parsed_day_option.value();
+                if (parsed_day < cMinParsedDay || parsed_day > cMaxParsedDay) {
+                    return ErrorCodeEnum::IncompatibleTimestampPattern;
+                }
+
+                timestamp_idx += cFieldLength;
+                date_type_representation = true;
+                break;
+            }
+            case 'e': {  // Space-padded day in month.
+                constexpr size_t cFieldLength = 2;
+                if (timestamp_idx + cFieldLength > timestamp.size()) {
+                    return ErrorCodeEnum::IncompatibleTimestampPattern;
+                }
+
+                auto const parsed_day_option{convert_padded_string_to_number(
+                        timestamp.substr(timestamp_idx, cFieldLength),
+                        ' '
+                )};
+                if (false == parsed_day_option.has_value()) {
+                    return ErrorCodeEnum::IncompatibleTimestampPattern;
+                }
+
+                parsed_day = parsed_day_option.value();
+                if (parsed_day < cMinParsedDay || parsed_day > cMaxParsedDay) {
+                    return ErrorCodeEnum::IncompatibleTimestampPattern;
+                }
+
+                timestamp_idx += cFieldLength;
+                date_type_representation = true;
+                break;
+            }
+            case 'a': {  // Abbreviated day in week.
+                auto const day_idx{YSTDLIB_ERROR_HANDLING_TRYX(find_first_matching_prefix(
+                        timestamp.substr(timestamp_idx),
+                        cAbbreviatedDaysOfWeek
+                ))};
+                timestamp_idx += cAbbreviatedDaysOfWeek.at(day_idx).length();
+                day_of_week_idx = static_cast<int>(day_idx);
+                date_type_representation = true;
+                break;
+            }
+            case 'p':
+            case 'H':
+            case 'k':
+            case 'I':
+            case 'l':
+            case 'M':
+            case 'S':
+            case '3':
+            case '6':
+            case '9':
+            case 'T':
+            case 'E':
+            case 'L':
+            case 'C':
+            case 'N':
+            case 'z':
+            case 'Z':
+            case '?':
+            case 'P':
+                return ErrorCodeEnum::FormatSpecifierNotImplemented;
+            case '\\': {
+                if ('\\' == timestamp[timestamp_idx]) {
+                    ++timestamp_idx;
+                    continue;
+                }
+                return ErrorCodeEnum::IncompatibleTimestampPattern;
+            }
+            default:
+                return ErrorCodeEnum::InvalidTimestampPattern;
+        }
+    }
+
+    // Do not allow trailing unmatched content.
+    if (false == (pattern_idx == pattern.size() && timestamp_idx == timestamp.size())) {
```

Comment:
```suggestion
    if (pattern_idx != pattern.size() || timestamp_idx != timestamp.size())) {
```
Nit: easier to read?